src/prism.c

```diff
@@ -6452,7 +6454,8 @@ lex_numeric(pm_parser_t *parser) {
     parser->integer_base = PM_INTEGER_BASE_FLAGS_DECIMAL;
 
     if (parser->current.end < parser->end) {
-        type = lex_numeric_prefix(parser);
+        bool seen_e = false;
```

Comment:
The variable name `seen_e` is derived from CRuby's implementation. If anyone has a better name, please suggest it to me.